calculator.js

```diff
@@ -148,31 +162,47 @@ function handlesReset(){
 
 function handlesDelete(){
     let digits = digitsOnScreen.innerHTML;
+
     if (digits[digits.length-1] === '.'){
         document.getElementById("decimalSeparator").classList.remove('dot-disabled');    
+    } else if (digits[digits.length-2] === '-'){
+        digits = digits.slice(0,-2);
+    } else {
+        digits = digits.slice(0,-1);
     }
-    digits = digits.slice(0,-1);
+    
     clearScreen();
     replaceOnScreen(digits);
 }
 
 function handlesEquals(){
+    if (!operator){
+        return;
+    }
     num2 = digitsOnScreen.innerHTML;
     num1 = parseToNum(num1);
     num2 = parseToNum(num2); 
     const result = calculateResult(operator, num1, num2);
     handlesReset();
     replaceOnScreen(result);
+    if (result.toString().includes('.')){
+        btnDecSeparator.className += ' dot-disabled';
+    }
     console.log(result);
 }
 
 function handlesDecSeparator(value){
-    addToScreen(value,'add');
-    btnDecSeparator.className += ' dot-disabled';
+    if (digitsOnScreen.innerHTML.includes('.') && operatorState){
+        replaceOnScreen(value);
+        btnDecSeparator.className += ' dot-disabled';
+    } else{
+        addToScreen(value);
+        btnDecSeparator.className += ' dot-disabled';
```

Comment:
if this instruction will happen in both scenarios it's better to put outside of the conditions.
```javascript
btnDecSeparator.className += ' dot-disabled';
```